### PR TITLE
Add initWithValue as a vuexfire mutation.

### LINKED ISSUE #30: Allow vuex strict mode
Hi!

Was trying to make this works on my project with no success. I have multiple modules on my store and have modules on different files, my code is like as follow

**store/index.js**
```
import VuexFire from 'vuexfire'
import * as actions from './actions'
import * as getters from './getters'

... // HERE I IMPORT SOME OTHERS MODULES

import raffles from './modules/raffles'

Vue.use(Vuex)
Vue.use(VuexFire)

const store = new Vuex.Store({
  strict: true, 
  actions,
  getters,
  modules: {
    ... // HERE I ADD THE OTHERS MODULES
    raffles
  },
  state: {
  },
  mutations: {
  }
})

export default store
```

**store/modules/raffles.js**
```
import VuexFire from 'vuexfire'

const state = {
  raffles: null
}

const mutations = VuexFire.moduleMutations('raffles')

export default {
  state,
  mutations
}
```

**app.js**
```
import Vue from 'vue'
import * as firebase from 'firebase'
import store from './store'
import { sync } from 'vuex-router-sync'
import App from './App'

// Initialize Firebase
var config = {
  apiKey: process.env.FIREBASE_API_KEY,
  authDomain: process.env.FIREBASE_AUTH_DOMAIN,
  databaseURL: process.env.FIREBASE_DATABASE_URL,
  storageBucket: process.env.FIREBASE_STORAGE_BUCKET,
  messagingSenderId: process.env.FIREBASE_MESSAGING_SENDER_ID
}
const firebaseApp = firebase.initializeApp(config)
const db = firebaseApp.database()

const app = new Vue({
  store,
  ...App,
  firebase: {
    raffles: db.ref('raffles')
  }
})

export { app, store, firebaseApp }
```

I don't know why i am getting this full error on the console:
```
[Vue warn]: Error in watcher "state" 
(found in root instance)
warn @ vue.common.js?e881:521
vue.common.js?e881:3027 Uncaught Error: [vuex] Do not mutate vuex store state outside mutation handlers.
    at assert (eval at <anonymous> (app.js:904), <anonymous>:214:27)
    at Vue$3.store._vm.$watch.deep (eval at <anonymous> (app.js:904), <anonymous>:755:5)
    at Watcher.run (eval at <anonymous> (app.js:796), <anonymous>:3017:19)
    at Watcher.update (eval at <anonymous> (app.js:796), <anonymous>:2991:10)
    at Dep.notify (eval at <anonymous> (app.js:796), <anonymous>:580:13)
    at Object.reactiveSetter [as raffles] (eval at <anonymous> (app.js:796), <anonymous>:810:11)
    at Object.initWithValue (eval at <anonymous> (app.js:1391), <anonymous>:475:17)
    at bindAsArray (eval at <anonymous> (app.js:1391), <anonymous>:196:15)
    at bind (eval at <anonymous> (app.js:1391), <anonymous>:180:6)
    at Vue$3.created (eval at <anonymous> (app.js:1391), <anonymous>:336:8)
```

Thanks and any helps will be appreciated.

### PR DESCRIPTION
**Pull request by holic-cl**
With this modification strict mode is supported, and this commit will
change compied version of vuexfire.js and vuexfire.min.js, adds strict mode to
TodoApp/index.html file, and removes initWithValue method from
utils/vuex.js file. This close #30.

### CODE REVIEW

**Comment by holic-cl** [comment]
i will try, first time i use vuex, and first time i use vue for a project, so i don't try any tool to make test yet.

I will try anyway.

**Comment by posva** [comment]
No worries. If you don't manage to write the test easily, simply remove the dist from the commit

File: dist/vuexfire.js [hunk 1]
```diff
@@ -488,6 +492,12 @@ return /******/ (function(modules) { // webpackBootstrap
 	    : mutation
 	}
 
+	exports.initModuleKeyInState = function initModuleKeyInState (state, module) {
+	  return state = module
```
Comment (posva) [inline dist/vuexfire.js]: no need to to set the state variable, it changes nothing anyway

File: dist/vuexfire.js [hunk 2]
```diff
@@ -488,6 +492,12 @@ return /******/ (function(modules) { // webpackBootstrap
 	    : mutation
 	}
 
+	exports.initModuleKeyInState = function initModuleKeyInState (state, module) {
```
Comment (posva) [inline dist/vuexfire.js]: I'd prefer something like `getModuleState`

File: src/plugin.js [hunk 3]
```diff
@@ -129,8 +130,15 @@ function bind (vm, fullKey, source) {
  */
 function bindAsArray (vm, fullKey, module, key, source, cancelCallback) {
   var state = vm.$store.state
+
+  // need to set up module keys in the state
+  utils.vuex.initModuleKeyInState(state, module)
```
Comment (posva) [inline src/plugin.js]: This shouldn't be here

File: src/plugin.js [hunk 4]
```diff
@@ -320,6 +328,10 @@ install.mutations[VUEXFIRE_ARRAY_MOVE] = function (state, payload) {
   array.splice(payload.newIndex, 0, array.splice(payload.index, 1)[0])
 }
 
+install.mutations[VUEXFIRE_INIT_WITH_VALUE] = function (state, payload) {
+  state[payload.key] = payload.value
```
Comment (posva) [inline src/plugin.js]: Here is where the method `getModuleState` should be called `getModuleState(state, module)[key] = value`

File: dist/vuexfire.js (hunk 2 above)
Comment (posva) [inline dist/vuexfire.js]: Commented on the dist file but I meant it on the utils file

**Comment by holic-cl** [comment]
I'm a bit confused on how the module and submodule have to be created.
This is how the state var is being created:
Object{items: [], user: Object{cart: Object{}}}

File: src/plugin.js [hunk 5]
```diff
@@ -129,8 +130,13 @@ function bind (vm, fullKey, source) {
...
   // set it as an array
-  utils.vuex.initWithValue(state, module, key, [])
+  vm.$store.commit(utils.vuex.getMutationName(module, VUEXFIRE_INIT_WITH_VALUE), {
+    key: key,
+    value: [],
+    module: module
+  })
 
   const onAdd = source.on('child_added', function (snapshot, prevKey) {
     const array = utils.vuex.get(state, module, key)
```
Comment (posva) [inline src/plugin.js]: You should actually create the array before the commit and use it as the value above

File: src/plugin.js [hunk 6]
```diff
@@ -129,8 +130,13 @@ function bind (vm, fullKey, source) {
  */
 function bindAsArray (vm, fullKey, module, key, source, cancelCallback) {
   var state = vm.$store.state
+
   // set it as an array
-  utils.vuex.initWithValue(state, module, key, [])
+  vm.$store.commit(utils.vuex.getMutationName(module, VUEXFIRE_INIT_WITH_VALUE), {
+    key: key,
+    value: [],
```
Comment (posva) [inline src/plugin.js]: Here we can pass on the `array`

**Comment by posva** [comment]
Hey, sorry if I felt too picky, I hope you didn't fly away 😅
The master branch now holds the v2, if you or anyone else wants to work on this again they can open a new PR against the v1 branch

Thanks a lot for taking your time to help out, I actually really appreciate it 🙂

**Comment by holic-cl** [comment]
You sorry me, i have been busy and i didn't back to work on this again.

I really appreciate your work.

**Comment by posva** [comment]
No worries. Thanks for the support 😄
Make sure to check the v2, it supports vuexfire out of the box 🙂

**Pull request closed without merging**